crates/core/src/val/record.rs

```diff
@@ -315,3 +331,94 @@ pub(crate) enum RecordType {
 	/// Represents an edge in a graph
 	Edge,
 }
+
+/// Metadata associated with a record
+///
+/// This struct contains optional metadata about a record, such as its type and
+/// aggregation statistics for materialized view records.
+/// The metadata is revisioned to ensure compatibility across different versions
+/// of the database.
+#[revisioned(revision = 2)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub(crate) struct Metadata {
+	/// The type of the record (e.g., Edge for graph edges)
+	record_type: Option<RecordType>,
+	/// Aggregation statistics for materialized view records
+	stats: Option<HashMap<String, FieldStats>>,
```

Comment:
Why is this an `Option<...>` what does `stats: None` mean that isn't just a lack of statistics and why not make that be an empty hashmap?